src/olympia/files/models.py

```diff
@@ -377,21 +377,25 @@ def get_localepicker(self):
         statsd.timing('files.extract.localepicker', (end * 1000))
         return res
 
-    @amo.cached_property
+    @property
     def webext_permissions(self):
-        try:
-            return {
-                name: WEBEXT_PERMISSIONS.get(name, Permission(name, '', ''))
-                for name in self._webext_permissions.permissions}
-        except WebextPermission.DoesNotExist:
-            return {}
+        return {
+            name: WEBEXT_PERMISSIONS.get(name, Permission(name, '', ''))
+            for name in self.webext_permissions_list}
 
-    @amo.cached_property
+    @property
     def webext_permissions_known(self):
         return {name: perm
                 for name, perm in self.webext_permissions.iteritems()
                 if name in WEBEXT_PERMISSIONS}
 
+    @amo.cached_property(writable=True)
+    def webext_permissions_list(self):
```

Comment:
Could you check `self.is_webextension` before accessing `self._webext_permissions` ? That way you avoid an useless database query for non-webextension files.

It's especially interesting here, because that's what the API uses, but it might also be useful in the other properties as well.